### PR TITLE
chore: add containerisation

### PR DESCRIPTION
**Pull request by dukenguyenxyz**
- Resolves https://github.com/PlaceOS/compiler/issues/17

### CODE REVIEW

File: Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    --disabled-password \
+    --gecos "" \
+    --home "/nonexistent" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    "${USER}"
+
+COPY shard.yml /app
+COPY shard.lock /app
```
Comment (caspiano) [inline Dockerfile.dev]:
```suggestion
````

File: Dockerfile.dev [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+RUN PLACE_COMMIT=$PLACE_COMMIT \
+    shards install --production
```
Comment (caspiano) [inline Dockerfile.dev]:
```suggestion
RUN shards install
```

File: Dockerfile.spec [hunk 3]
```diff
@@ -0,0 +1,31 @@
+FROM crystallang/crystal:0.35.1-alpine
+
+WORKDIR /app
+
+# Set the commit through a build arg
+ARG PLACE_COMMIT="DEV"
+
```
Comment (caspiano) [inline Dockerfile.spec]:
```suggestion
```

File: test [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+# initialise trap to call trap_ctrlc function
+# when signal 2 (SIGINT) is received
+trap "trap_ctrlc" 2
+
+docker-compose build
+
+docker-compose run --rm compiler /app/entrypoint $@
+
+docker-compose down
```
Comment (stakach) [inline test]: worth adding the trailing new line whitespace

File: .gitignore [hunk 5]
```diff
@@ -1,5 +1,4 @@
 doc
-shard.lock
```
Comment (caspiano) [inline .gitignore:2]:
```suggestion
shard.lock
```

no reason to have a lock file in a lib

File: README.md [hunk 6]
```diff
@@ -3,3 +3,8 @@
 [![Build Status](https://travis-ci.com/PlaceOS/compiler.svg?branch=master)](https://travis-ci.com/PlaceOS/compiler)
 
 Collection of PlaceOS Driver compilation tooling.
+
+## Development
+
+To spec, run:
+`./test`
```
Comment (caspiano) [inline README.md]:
```suggestion
- `./test` to run spec suite
- `./test --watch` to run spec suite on file changes
```

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,29 @@
+FROM crystallang/crystal:0.35.1-alpine
+
+WORKDIR /app
+
+# Set the commit through a build arg
+ARG PLACE_COMMIT="DEV"
+
+# Install the latest version of LibSSH2, ping
+RUN apk add --no-cache libssh2 libssh2-dev iputils
```
Comment (caspiano) [inline Dockerfile]:
```suggestion
 RUN apk add --no-cache libssh2 libssh2-dev libssh2-static
 ```

**Pull request merged**